otodom/task_1/mk/otodom.py

```diff
@@ -0,0 +1,165 @@
+            number = bs.find(
+                "div", {"data-testid": "table-value-rooms_num"}
+            ).text.replace(",", ".")
+            listing["rooms"] = int(float(re.sub(r"[^.0-9]", "", number)))
+            number = bs.find("div", {"data-testid": "table-value-area"}).text.replace(
+                ",", "."
+            )
+            listing["area"] = int(float(re.sub(r"[^.0-9]", "", number)))
+            listing["estate_agency"] = bs.find(
+                "div", {"data-testid": "table-value-advertiser_type"}
+            ).text
+
+            return listing
+        return None
+
+    def scrap_listings(self, url, check_all_pages=False):
+        bs = self.getPage(url)
+
+        if bs is not None:
+            listings_url = set()
+            listing_json = []
+
+            number_of_pages = 1
+            if check_all_pages:
+                page_numeration = bs.find_all(
+                    "a", {"class": "eo9qioj1 css-5tvc2l edo3iif1"}
+                )
+                number_of_pages = max([int(num.text) for num in page_numeration])
+
+            for page_number in range(1, number_of_pages + 1):
+                print(page_number)
+                listing_links = self.getLinks(url + "?page={}".format(page_number))
+                # listing_links = self.getLinks(url)
+                # promoted ads
+                for listing_url in listing_links[0]:
+                    if listing_url not in listings_url:
+                        listings_url.add(listing_url)
+                        listing_json.append(self.getListing(listing_url, promoted=True))
+
+                # organic ads
+                for listing_url in listing_links[1]:
+                    if listing_url not in listings_url:
+                        listings_url.add(listing_url)
+                        listing_json.append(
+                            self.getListing(listing_url, promoted=False)
+                        )
+
+            with open("otodom_listing.json", "w", encoding="utf-8") as json_file:
+                json.dump(listing_json, json_file, ensure_ascii=False, indent=2)
+
+    def generate_url(self):
+        with open("otodom_settings.json") as f:
+            data = json.load(f)
+            url = data["base_url"] + "pl/wyniki"
+
+            if data["only_for_sale"]:
+                url += "/sprzedaz"
+
+            if data["only_for_rent"]:
+                url += "/wynajem"
+                url += "/" + data["property_type"] + "/"
+            if len(data["province"]) > 0:
+                url += data["province"] + "/" + data["city"] + "?"
+            else:
+                url += "cala-polska?"
+
+            url += "limit=36"
+
+            if len(data["price_min"]) > 0:
+                url += "&priceMin=" + data["price_min"]
+
+            if len(data["price_max"]) > 0:
+                url += "&priceMax=" + data["price_max"]
+
+            url += "&by=LATEST&direction=DESC&viewType=listing"
```

Comment:
all of those are so-called parameters (starting from `?` in `cala-polska?`).  when making a request you can add them as a dictionary.
```
params = {
    "price_min": "100",
    "price_max": "200,
    ...
}
requests.get(url=url, params=params)
```
these are going to be automatically let's say "parsed"
the final url would look like `https://github.com?price_min=100&price_max=200`